packages/server/src/shared/internal/serialize.ts

```diff
@@ -18,6 +18,8 @@ type IsAny<T> = 0 extends T & 1 ? true : false;
 // support it as both a Primitive and a NonJsonPrimitive
 type JsonReturnable = JsonPrimitive | undefined;
 
+type IsRecord<T> = keyof WithoutIndexSignature<T> extends never ? true : false;
```

Comment:
I'll change this to have an object check as well. Even though no tests fail, this is still too broad